dags/analyzer_helper/tests/integration/test_discord_transform_raw_data.py

```diff
@@ -0,0 +1,158 @@
+import unittest
+from datetime import datetime
+
+from dags.analyzer_helper.discord.discord_transform_raw_data import DiscordTransformRawData
+
+
+class TestDiscordTransformRawData(unittest.TestCase):
+
+    def setUp(self):
+        self.transformer = DiscordTransformRawData()
+        self.platform_id = 'discord_platform'
+        self.period = datetime(2023, 1, 1)
+
+    def test_transform_data_with_replied(self):
+        raw_data = [
+            {
+                "author": "user123",
+                "replied": "user789",
+                "messageId": "msg123",
+                "channelId": "channel456",
+                "channelName": "general",
+                "threadId": "thread123",
+                "threadName": "discussion",
+                "isGeneratedByWebhook": False
+            }
+        ]
+
+        expected_result = [
+            {
+                "author_id": "user123",
+                "date": self.period,
+                "source_id": self.platform_id,
+                "metadata": {
```

Comment:
The `source_id` would be the `messageId` and please update the metadata based on the previous comments.